dmarc/lookup.go

```diff
@@ -24,9 +24,21 @@ func IsTempFail(err error) bool {
 
 var ErrNoPolicy = errors.New("dmarc: no policy found for domain")
 
+type txtLookupFunc func(domain string) ([]string, error)
+
 // Lookup queries a DMARC record for a specified domain.
 func Lookup(domain string) (*Record, error) {
-	txts, err := net.LookupTXT("_dmarc." + domain)
+	return LookupWithOption(domain, nil)
+}
+
+func LookupWithOption(domain string, txtLookup txtLookupFunc) (*Record, error) {
```

Comment:
I'd prefer to introduce a `LookupOptions` struct here too, in case we want to add more options in the future.

Reply:
added